src/kalman/kalmanFilter.cpp

```diff
@@ -27,40 +34,59 @@ void kalmanFilter::kalmanUpdate() {
     }
     ball.kalmanUpdateK();
     ball.kalmanUpdateX();
+
+
 }
 
 // if we get a new frame we update our observations
-void kalmanFilter::newFrame(const roboteam_msgs::DetectionFrame &msg) {
-    double timeCapture = msg.t_capture;
+void kalmanFilter::newFrame(const roboteam_proto::SSL_DetectionFrame &msg) {
+  std::lock_guard<std::mutex> lock(filterMutex);
+
+  double timeCapture = msg.t_capture();
     lastFrameTime = timeCapture;
-    uint cameraID = msg.camera_id;
-    for (const roboteam_msgs::DetectionRobot robot : msg.us) {
-        ourBots[robot.robot_id].kalmanUpdateZ(robot, timeCapture, cameraID);
+    uint cameraID = msg.camera_id();
+    for (const roboteam_proto::SSL_DetectionRobot& robot : msg.robots_yellow()) {
+        ourBots[robot.robot_id()].kalmanUpdateZ(robot, timeCapture, cameraID);
     }
-    for (const roboteam_msgs::DetectionRobot robot : msg.them) {
-        theirBots[robot.robot_id].kalmanUpdateZ(robot, timeCapture, cameraID);
+    for (const roboteam_proto::SSL_DetectionRobot& robot : msg.robots_blue()) {
+        theirBots[robot.robot_id()].kalmanUpdateZ(robot, timeCapture, cameraID);
     }
-    for (const roboteam_msgs::DetectionBall detBall : msg.balls) {
+    for (const roboteam_proto::SSL_DetectionBall& detBall : msg.balls()) {
         ball.kalmanUpdateZ(detBall, timeCapture, cameraID);
     }
+
 }
 
 //Creates a world message with the currently observed objects in it
-roboteam_msgs::World kalmanFilter::getWorld() {
-    roboteam_msgs::World world;
-    world.time = lastFrameTime;
+roboteam_proto::World kalmanFilter::getWorld() {
+  std::lock_guard<std::mutex> lock(filterMutex);
+
+  roboteam_proto::World world;
+    world.set_time(lastFrameTime);
     for (const auto& kalmanOurBot : ourBots){
         if (kalmanOurBot.getExistence()){
-            world.us.push_back(kalmanOurBot.as_message());
+            world.mutable_yellow()->Add(kalmanOurBot.as_message());
         }
     }
     for (const auto& kalmanTheirBot : theirBots){
         if (kalmanTheirBot.getExistence()){
-            world.them.push_back(kalmanTheirBot.as_message());
+            world.mutable_blue()->Add(kalmanTheirBot.as_message());
         }
     }
-    world.ball=ball.as_ball_message();
-    return world;
+
+    roboteam_proto::WorldBall worldBall = ball.as_ball_message();
+    world.mutable_ball()->CopyFrom(worldBall);
```

Comment:
I am not sure if these 2 lines of code make one or 2 objects of the worldBall in total.